# Author : Petros S (petross404@gmail.com)
# Date : 2017-06-16

# Patch GammaRay's top-level cmakelists.txt in order to delete automagical deps for Portage.
# GammaRay uses a lot of optional Qt packages based on whether these packages are
# present in the system. So a USE flag will determine if GammaRay will search and link
# to these packages.

--- gammaray-2.8.0/CMakeLists.txt	2017-06-16 16:05:44.427083989 +0300
+++ gammaray-2.8.0/CMakeLists.txt	2017-06-16 16:14:27.517235711 +0300
@@ -322,25 +322,6 @@
   set_package_properties(Qt5Core PROPERTIES TYPE REQUIRED)
   find_package(Qt5 NO_MODULE REQUIRED COMPONENTS Gui Network Xml)
   find_package(Qt5 NO_MODULE QUIET OPTIONAL_COMPONENTS
-    3DExtras
-    3DInput
-    3DLogic
-    3DRender
-    3DQuick
-    Bluetooth
-    Concurrent
-    Positioning
-    PrintSupport
-    Qml
-    Quick
-    QuickWidgets
-    Svg
-    Script
-    ScriptTools
-    Test
-    WebKit
-    WebKitWidgets
-    Widgets
     WaylandCompositor
   )
   # Find these 'exotic' Qt modules without using find_package(... COMPONENTS)
@@ -368,15 +349,14 @@
   endif()
 
   # starting with 5.6, the Designer module needs CMake 3.0 or higher
-  if(Qt5Core_VERSION VERSION_LESS 5.6 OR NOT CMAKE_VERSION VERSION_LESS 3.0)
-    find_package(Qt5 NO_MODULE QUIET OPTIONAL_COMPONENTS Designer)
-  endif()
+  
   if(GAMMARAY_BUILD_UI) # widgets are required for the UI
     find_package(Qt5 NO_MODULE REQUIRED COMPONENTS Widgets)
     # 3D Widget inspector requires Qt3D 5.7
     if(Qt5Core_VERSION VERSION_EQUAL 5.7 OR Qt5Core_VERSION VERSION_GREATER 5.7)
       if(${Qt53DRender_FOUND} AND ${Qt53DInput_FOUND} AND ${Qt53DLogic_FOUND} AND ${Qt53DQuick_FOUND})
-        set(GAMMARAY_WITH_WIDGET3D TRUE)
+        set(GAMMARAY_WITH_WIDGET3D FALSE)
+		message(STATUS "Not building GammaRay Widget 3D")
       else()
         message(STATUS "Not building GammaRay Widget 3D")
       endif()
@@ -408,11 +388,11 @@
 
   include("cmake/ECMQt4To5Porting.cmake")
 
-  set(HAVE_QT_CONCURRENT ${Qt5Concurrent_FOUND})
+  set(HAVE_QT_CONCURRENT OFF)
   set(HAVE_QT_WIDGETS ${Qt5Widgets_FOUND})
-  set(HAVE_QT_SVG ${Qt5Svg_FOUND})
-  set(HAVE_QT_SCXML ${Qt5Scxml_FOUND})
-  set(HAVE_QT_DESIGNER ${Qt5Designer_FOUND})
+  set(HAVE_QT_SVG OFF)
+  set(HAVE_QT_SCXML OFF)
+  set(HAVE_QT_DESIGNER OFF)
 
   if(Qt5PrintSupport_FOUND)
     cmake_push_check_state()
@@ -423,8 +403,8 @@
       set(HAVE_QT_PRINTSUPPORT ON)
     endif()
   endif()
-  set(HAVE_QT_WEBKIT ${Qt5WebKit_FOUND})
-  set(HAVE_QT_WEBKIT1 ${Qt5WebKitWidgets_FOUND})
+  set(HAVE_QT_WEBKIT OFF)
+  set(HAVE_QT_WEBKIT1 OFF)
 
   # TODO: Remove me once fixed in ECM module
   # more hacks: find qpa/... includes
@@ -465,9 +445,7 @@
   set(HAVE_QT_CONCURRENT true)
   set(HAVE_QT_WIDGETS true)
   set(HAVE_QT_SVG true)
-  if(QT_QTDESIGNER_FOUND)
-    set(HAVE_QT_DESIGNER true)
-  endif()
+
   set(HAVE_QT_PRINTSUPPORT true)
   set(HAVE_QT_WEBKIT ${QT_QTWEBKIT_FOUND})
   set(HAVE_QT_WEBKIT1 ${QT_QTWEBKIT_FOUND})
@@ -540,7 +518,6 @@
 
 add_feature_info("QtScript debugger" QT_QTSCRIPTTOOLS_FOUND "Requires QtScript and QtScriptTools.")
 add_feature_info("Web inspector" HAVE_QT_WEBKIT1 "Requires QtWebKit.")
-add_feature_info("Widget .ui file export" HAVE_QT_DESIGNER "Requires QtDesigner library.")
 add_feature_info("3D Widget Inspector" GAMMARAY_WITH_WIDGET3D "Requires Qt5 >= 5.7, Qt3D and QtQuick Controls")
 
 #
@@ -902,5 +879,5 @@
 feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
 
 if(NOT Qt5Core_FOUND AND NOT Qt4_FOUND)
-  message(FATAL_ERROR "GammaRay needs either Qt5 or Qt4!")
+  message(FATAL_ERROR "GammaRay needs either Qt5 or Qt4! Only Qt5 is supported in the Gentoo ebuild.")
 endif()
